node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/DefaultEnvRewriter.java

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task.util;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import static com.yahoo.yolean.Exceptions.uncheck;
+import static java.nio.file.StandardCopyOption.ATOMIC_MOVE;
+
+/**
+ * Rewrites default-env.txt files.
+ *
+ * @author bjorncs
+ */
+public class DefaultEnvRewriter {
+
+    private final Map<String, Operation> operations = new TreeMap<>();
+    private final Path defaultEnvFile;
+
+    public DefaultEnvRewriter(Path defaultEnvFile) {
+        this.defaultEnvFile = defaultEnvFile;
+    }
+
+    public DefaultEnvRewriter addOverride(String name, String value) {
+        return addOperation("override", name, value);
+    }
+
+    public DefaultEnvRewriter addFallback(String name, String value) {
+        return addOperation("fallback", name, value);
+    }
+
+    public DefaultEnvRewriter addUnset(String name) {
+        return addOperation("unset", name, null);
+    }
+
+    private DefaultEnvRewriter addOperation(String action, String name, String value) {
+        if (operations.containsKey(name)) {
+            throw new IllegalArgumentException(String.format("Operation on variable '%s' already added", name));
+        }
+        operations.put(name, new Operation(action, name, value));
+        return this;
+    }
+
+    public boolean converge() {
+        List<String> defaultEnvLines = uncheck(() -> Files.readAllLines(defaultEnvFile));
```

Comment:
Is the file always guaranteed to exist? Maybe better with
```suggestion
        List<String> defaultEnvLines = ifExists(() -> Files.readAllLines(defaultEnvFile)).orElseGet(List::of);
```

Reply:
It's part of the Vespa installation. I'll change the code to handle missing file anyway.